src/reader.rs

```diff
@@ -73,20 +70,38 @@ pub trait CarReader {
         (*cid, fs_ipld).try_into()
     }
 
+
+    fn search_file_cid_inner(&mut self, searchq: &mut VecDeque<Cid>, f: &str) -> Result<Cid, CarError> {
+        while let Some(root_cid) = searchq.pop_front() {
+            let fs_ipld = self.ipld(&root_cid)?;
```

Comment:
To improve performance, we can batch the calls to self.ipld() and avoid unnecessary repeated calls?

Thinking this can be achieved by caching the results of self.ipld() in a HashMap and retrieving them from the cache when a CID has already been processed.

maybe something like? 

```rust
let fs_ipld = *ipld_map.entry(root_cid.clone())
    .or_insert_with(|| self.ipld(&root_cid).unwrap());
```

Reply:
Yes, It's necessary for improve performance.